cmd/server/app/options/options.go

```diff
@@ -86,7 +87,8 @@ func (o *ProxyRunOptions) Flags() *pflag.FlagSet {
 	flags.UintVar(&o.AgentPort, "agent-port", o.AgentPort, "Port we listen for agent connections on.")
 	flags.UintVar(&o.AdminPort, "admin-port", o.AdminPort, "Port we listen for admin connections on.")
 	flags.UintVar(&o.HealthPort, "health-port", o.HealthPort, "Port we listen for health connections on.")
-	flags.DurationVar(&o.KeepaliveTime, "keepalive-time", o.KeepaliveTime, "Time for gRPC server keepalive.")
+	flags.DurationVar(&o.KeepaliveTime, "keepalive-time", o.KeepaliveTime, "Time for gRPC agent server keepalive.")
+	flags.DurationVar(&o.KeepaliveTime, "master-keepalive-time", o.MasterKeepaliveTime, "Time for gRPC master server keepalive.")
```

Comment:
Your getting your default from the MasterKeepaliveTime variable but your then writing it back to the KeepaliveTime variable and over-writing whatever was set by the "keepalive-time" flag.

Reply:
updated